Add docs about how to use private repositories (#854)
kubeapps/kubeapps: docs/user/private-app-repository.md

To add your private repository go to `Configuration > App Repositories` in Kubeapps and click on "Add App Repository". You will need to add your repository using the Kubernetes DNS name for the ChartMuseum service. This will be `<release_name>-chartmuseum.<namespace>:8080`:

<img src="../img/chartmuseum-repository.png" alt="ChartMuseum App Repository" width="600px">

Once you create the repository you can click on the link for the specific repository and you will be able to deploy your own applications using Kubeapps.

### ChartMuseum: Authentication/Authorization

It is possible to configure ChartMuseum to use authentication with two different mechanism:

- Using HTTP [basic authentication](https://chartmuseum.com/docs/#basic-auth) (user/password). To use this feature, it's needed to:
  - Specify the parameters `secret.AUTH_USER` and `secret.AUTH_PASS` when deploying the ChartMuseum.
  - Set as Authorization Header a `Basic` auth header using as value the base64 codification of the string `user:password`. For example, for the user "Aladdin" and password "open sesame", it would use the following header field: `Basic QWxhZGRpbjpvcGVuIHNlc2FtZQ==`

> - Using a [JWT token](https://github.com/chartmuseum/auth-server-example). Once you obtain a valid token you can set it in the Authorization Header field of the App Repository form. Note that in this case it will be prefixed with `Bearer` . For example: `Bearer UVd4aFpHUnBianB2Y0dWdUlIT=`.

Reviewer: extra space
```suggestion
- Using a [JWT token](https://github.com/chartmuseum/auth-server-example). Once you obtain a valid token you can set it in the Authorization Header field of the App Repository form. Note that in this case it will be prefixed with `Bearer`. For example: `Bearer UVd4aFpHUnBianB2Y0dWdUlIT=`.
```

## Artifactory

JFrog Artifactory is a Repository Manager supporting all major packaging formats, build tools and CI servers.

> **Note**: In order to use the Helm repository feature, it's necessary to use an Artifactory Pro account.

> To use Artifactory with Kubeapps the first thing is adding the JFrog repository to Kubeapps. Go to `Configuration > App Repositories` and add their repository:

Reviewer: ```suggestion
To install Artifactory with Kubeapps first add the JFrog repository to Kubeapps. Go to `Configuration > App Repositories` and add their repository:
```

<img src="../img/jfrog-repository.png" alt="JFrog repository" width="400px">

> Then click on the JFrog repository and deploy Artifactory. For detailed installation instructions, check its [README](https://github.com/jfrog/charts/tree/master/stable/artifactory). If you don't have any further requirement, the values by default will work.

Reviewer: ```suggestion
Then click on the JFrog repository and deploy Artifactory. For detailed installation instructions, check its [README](https://github.com/jfrog/charts/tree/master/stable/artifactory). If you don't have any further requirement, the default values will work.
```

> When deployed, during the first login, select "Helm" to initialize a repository:

Reviewer: what do you mean "during the first login", is that the setup wizard?

Author: yep

<img src="../img/jfrog-wizard.png" alt="JFrog repository" width="600px">

By default, Artifactory creates a chart repository called `helm`. That is the one you can use to store your applications.

### Artifactory: Upload a chart

> First, it's needed to obtain the user and password of the Helm repository. To obtain it, click in the `helm` repository and, in the `Set Me Up` menu, introduce your password. After that you will be able to see the user and password.

Reviewer: ```suggestion
First, you will need to obtain the user and password of the Helm repository. To do so, click on the `helm` repository and in the `Set Me Up` menu enter your password. After that you will be able to see the repository user and password.
```

Once you have done that, you will be able to upload a chart:

```
$ curl -u{USER}:{PASSWORD} -T /path/to/chart.tgz "http://{REPO_URL}/artifactory/helm/"
```

### Artifactory: Configure the repository in Kubeapps

To be able able to access private charts with Kubeapps first you need to generate a token. You can do that with the Artifactory API:

```
curl -u{USER}:{PASSWORD} -XPOST "http://{REPO_URL}/artifactory/api/security/token?expires_in=0" -d "username=kubeapps" -d "scope=member-of-groups:readers"
{
  "scope" : "member-of-groups:readers api:*",
  "access_token" : "TOKEN CONTENT",
  "token_type" : "Bearer"
}
```

> With that you have created a token with read-only permissions. Go to the `Configuration > App Repositories` menu and add your personal repository:

Reviewer: ```suggestion
The above command creates a token with read-only permissions. Now you can go to the `Configuration > App Repositories` menu and add your personal repository:
```

<img src="../img/jfrog-custom-repo.png" alt="JFrog custom repository" width="400px">

> After submitting the repository, you will be able to click in the new repository and see the chart uploaded in the previous step.

Reviewer: ```suggestion
After submitting the repository, you will be able to click on the new repository and see the chart you uploaded in the previous step.
```
